ipfs-3-to-4/migration/migration.go

```diff
@@ -387,6 +391,38 @@ func transferBlocks(flatfsdir string) error {
 
 	fmt.Println()
 
+	err := cleanEmptyDirs(flatfsdir)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	return nil
+}
+
+func cleanEmptyDirs(dir string) error {
+	children, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range children {
+		if !c.IsDir() {
+			continue
+		}
+
+		cdir := filepath.Join(dir, c.Name())
+		blocks, err := ioutil.ReadDir(cdir)
```

Comment:
@whyrusleeping the file you pointed me to only contains an implementation for `readdirnames`, `readdir` is implemeted in `os/file_unix.go` and it actually calls `Readdirnames` first and than calls `lstat` for each entry.  I don't see how it could be implemented any other way because of how the related system calls on unix work.